src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -0,0 +1,222 @@
+package net.snowflake.client.jdbc;
+
+import static org.junit.Assert.*;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import net.snowflake.client.ConditionalIgnoreRule;
+import net.snowflake.client.RunningOnGithubAction;
+import net.snowflake.client.category.TestCategoryStructuredType;
+import net.snowflake.client.core.structs.SnowflakeObjectTypeFactories;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(TestCategoryStructuredType.class)
+public class ResultSetStructuredTypesLatestIT extends BaseJDBCTest {
+  private final String queryResultFormat;
+
+  public ResultSetStructuredTypesLatestIT() {
+    this("JSON");
+  }
+
+  protected ResultSetStructuredTypesLatestIT(String queryResultFormat) {
+    this.queryResultFormat = queryResultFormat;
+  }
+
+  public Connection init() throws SQLException {
+    Connection conn = BaseJDBCTest.getConnection(BaseJDBCTest.DONT_INJECT_SOCKET_TIMEOUT);
+    Statement stmt = conn.createStatement();
+    stmt.execute("alter session set ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE = true");
+    stmt.execute("alter session set IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE = true");
+    stmt.execute("alter session set jdbc_query_result_format = '" + queryResultFormat + "'");
+    stmt.close();
+    return conn;
+  }
+
+  public static class SimpleClass implements SQLData {
```

Comment:
do we really wants to have it as a inner class?

Reply:
extracted